app/services/model_updater.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+# Applies updates Cocina Models
+class ModelUpdater
+  # @param [Cocina::Models::DRO] model the orignal state of the model
+  # @param [Hash<Symbol,String>] attributes the values to update. Pass nil as a value to clear the current field.
+  # @return [Cocina::Models::DRO] the model with updates applied
+  def self.update(model, attributes)
```

Comment:
For your consideration: I find the extra indirection of self.update(<with attributes>) confusing. Without reading the code, it isn't immediately apparent which attributes it will accept and how they are handled.

I'd suggest that more focused methods would make updates clearer. For example self.update_catkey().

Reply:
I've split those parts out into a ChangeSet class.  This is the pattern used by Valkyrie (https://github.com/samvera/valkyrie/wiki/ChangeSets-and-Dirty-Tracking)

Reply:
Thanks -- this pattern is clearer to me. Does this belong in cocina-models or dor-services-client so that it is available to all apps that might want to update models?

Reply:
I don't think so, the change-set is tightly coupled to the argo interactions.